File: Examples/test-suite/python/cpp11_move_only_runme.py
from cpp11_move_only import *

# Output
Counter.reset_counts()
mo = MoveOnly.create()
del mo
Counter.check_counts(1, 0, 0, 2, 0, 3)

Counter.reset_counts()
mc = MovableCopyable.create()
del mc
Counter.check_counts(2, 1, 0, 0, 1, 3)

# Move semantics not used
Counter.reset_counts()
mc = MovableCopyable.createConst()
del mc
Counter.check_counts(2, 1, 1, 0, 0, 3)

# Input static method
Counter.reset_counts()
mc = MovableCopyable(222)
Counter.check_counts(1, 0, 0, 0, 0, 0)
MovableCopyable.take(mc)
Counter.check_counts(2, 0, 1, 1, 0, 2)
del mc
Counter.check_counts(2, 0, 1, 1, 0, 3)

# Input global function
Counter.reset_counts()
mc = MovableCopyable(333)
Counter.check_counts(1, 0, 0, 0, 0, 0)
global_take_movable_copyable(mc)
Counter.check_counts(2, 0, 1, 1, 0, 2)
del mc
Counter.check_counts(2, 0, 1, 1, 0, 3)

# Input instance method
Counter.reset_counts()
imt = InstanceMethodsTester()
mc = MovableCopyable(444)
Counter.check_counts(1, 0, 0, 0, 0, 0)
imt.instance_take_movable_copyable(mc)
Counter.check_counts(2, 0, 1, 1, 0, 2)
del mc
Counter.check_counts(2, 0, 1, 1, 0, 3)

# Input constructor
Counter.reset_counts()
mc = MovableCopyable(555)
Counter.check_counts(1, 0, 0, 0, 0, 0)
ct = ConstructorTester(mc)
Counter.check_counts(2, 0, 1, 1, 0, 2)
del mc
Counter.check_counts(2, 0, 1, 1, 0, 3)

File: Examples/test-suite/javascript/cpp11_move_only_runme.js
var cpp11_move_only = require("cpp11_move_only");

// Output
/* await */(cpp11_move_only.Counter.reset_counts());
var mo = /* await */(cpp11_move_only.MoveOnly.create());
/* await */(cpp11_move_only.Counter.check_counts(1, 0, 0, 2, 0, 2));

/* await */((cpp11_move_only.Counter.reset_counts()));
var mo = /* await */(cpp11_move_only.MovableCopyable.create());
/* await */((cpp11_move_only.Counter.check_counts(2, 1, 0, 0, 1, 2)));

// Move semantics not used
/* await */(cpp11_move_only.Counter.reset_counts());
var mo = /* await */(cpp11_move_only.MovableCopyable.createConst());
/* await */(cpp11_move_only.Counter.check_counts(2, 1, 1, 0, 0, 2));

// Input static method
/* await */(cpp11_move_only.Counter.reset_counts());
var mc = new cpp11_move_only.MovableCopyable(222);
/* await */(cpp11_move_only.Counter.check_counts(1, 0, 0, 0, 0, 0));
/* await */(cpp11_move_only.MovableCopyable.take(mc));
/* await */(cpp11_move_only.Counter.check_counts(2, 0, 1, 1, 0, 2));

// Input global function
/* await */(cpp11_move_only.Counter.reset_counts());
var mc = new cpp11_move_only.MovableCopyable(333);
/* await */(cpp11_move_only.Counter.check_counts(1, 0, 0, 0, 0, 0));
/* await */(cpp11_move_only.global_take_movable_copyable(mc));
/* await */(cpp11_move_only.Counter.check_counts(2, 0, 1, 1, 0, 2));

// Input instance method
/* await */(cpp11_move_only.Counter.reset_counts());
var imt = new cpp11_move_only.InstanceMethodsTester();
var mc = new cpp11_move_only.MovableCopyable(444);
/* await */(cpp11_move_only.Counter.check_counts(1, 0, 0, 0, 0, 0));
/* await */(imt.instance_take_movable_copyable(mc));
/* await */(cpp11_move_only.Counter.check_counts(2, 0, 1, 1, 0, 2));

// Input constructor
/* await */(cpp11_move_only.Counter.reset_counts());
var mc = new cpp11_move_only.MovableCopyable(555);
/* await */(cpp11_move_only.Counter.check_counts(1, 0, 0, 0, 0, 0));
var ct = new cpp11_move_only.ConstructorTester(mc);
/* await */(cpp11_move_only.Counter.check_counts(2, 0, 1, 1, 0, 2));
